### ISSUE TITLE
TS 2.2: Indexers break type safety, affects many typings

### ISSUE THREAD

**Issue by arogg** [DefinitelyTyped/DefinitelyTyped#16316]
I just stumbled accross this because I wondered why the jquery typings would allow something like `$('#myid').fadeOutttt('slow') `which is not an error.

As it turns out TS 2.2 allows for dotted access for indexers.

This means when you add `[index: string]: any;` to any interface, type safety of the entire interface is **compromised**, meaning:

1) the only way to avoid typos is to use **autocompletion** help. The compiler will not report errors on typos!
2) when a type definition with indexer changes, the user will **not** be notified by the compiler that the interface has changed (e.g. method removed or signature changed).

In short: an indexer muddies its entire interface. It degrades the quality of many type definitions on this platform (e.g. jquery) that use indexers. 

2 solutions I see:

1) Please tell the typescript team this change is unacceptable and must change. If dotted access should be allowed, tell them to change their approach to it. We are the user base. Together we are stronger. A compiler switch is one way.
2) remove any indexers from typings that provide an alternative way to use the indexer, e.g. a `get(...)` method.

Any further ideas?

See https://github.com/Microsoft/TypeScript/issues/12596 for discussion

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
